From 688069175e805af756fa562873171d0a3fc4abd4 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:23 -0400
Subject: [ethernet] i40e: Remove a redundant filter addition

Message-id: <1409658042-22689-23-git-send-email-sassmann@redhat.com>
Patchwork-id: 90216
O-Subject: [RHEL7.1 PATCH 022/281] i40e: Remove a redundant filter addition
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

Remove a redundant filter addition to stop FW complaints about a redundant
filter removal.

Change-ID: I22bef6b682bd8d43432557e6e2b3e73ffb27b985
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 6982d429a9194e5069c5249e751422def87658a6)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 46b3507..50d0032 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -1965,11 +1965,14 @@ static int i40e_vlan_rx_add_vid(struct net_device *netdev,
 
  netdev_info(netdev, "adding %pM vid=%d\n", netdev->dev_addr, vid);
 
- /* If the network stack called us with vid = 0, we should
-  * indicate to i40e_vsi_add_vlan() that we want to receive
-  * any traffic (i.e. with any vlan tag, or untagged)
+ /* If the network stack called us with vid = 0 then
+  * it is asking to receive priority tagged packets with
+  * vlan id 0.  Our HW receives them by default when configured
+  * to receive untagged packets so there is no need to add an
+  * extra filter for vlan 0 tagged packets.
   */
- ret = i40e_vsi_add_vlan(vsi, vid ? vid : I40E_VLAN_ANY);
+ if (vid)
+  ret = i40e_vsi_add_vlan(vsi, vid);
 
  if (!ret && (vid < VLAN_N_VID))
   set_bit(vid, vsi->active_vlans);
-- 
1.7.1